openlabcmd/openlabcmd/zk.py

```diff
@@ -206,37 +217,116 @@ def delete_node(self, node_name):
         self.client.delete(path, recursive=True)
 
     @_client_check_wrapper
-    def list_services(self):
+    def list_services(self, node_name_filter=None, node_role_filter=None):
+        """
+        List the services in the HA deployment.
+        :param node_name_filter: The node filter.
+        :type node_name_filter: list or string.
```

Comment:
does the `--role` and `--node` arguments can be parsed to a list currently ?

Reply:
I can't see any improvement except simpler input format?

Reply:
may you can add `action='append', default=[]` to the arguments

Reply:
I don't think merge different filters into once arg is a good way. It's not comfortable for user experience IMO. Take `openstack server list` for example, each filter use the arg itself.

Reply:
emm, I think i misunderstand your message before.  See new PS please

Reply:
maybe you don't need L229~L240 if you added `default=[]` in the arguments

Reply:
openlabcmd is not only used for operator, but also for keepalived or some other manage tools in the future.  L229-L240 can make the input more flexible.

Reply:
OK,